test/functional/plugin/lsp_spec.lua

```diff
@@ -876,6 +853,56 @@ describe('LSP', function()
       }, buf_lines(1))
     end)
 
+    describe('edits with the same start position', function()
+      -- Spec: https://microsoft.github.io/language-server-protocol/specifications/specification-3-14/#textedit-1
+
+      -- Spec: If multiple inserts have the same position, the order in the
+      -- array defines the order in which the inserted strings appear in the
+      -- resulting text.
+      it('applies inserts in the order of the array', function()
+        local edits = {
+          make_edit(0, 18, 0, 18, {","});
+          make_edit(0, 18, 0, 18, {" after"});
+          make_edit(0, 18, 0, 18, {"\nnewline"});
+        }
+        exec_lua('vim.lsp.util.apply_text_edits(...)', edits, 1)
+        eq({
+          'First line of text, after';
+          'newline';
+          'Second line of text';
+          'Third line of text';
+          'Fourth line of text';
+          'å å ɧ 汉语 ↥ 🤦 🦄';
+        }, buf_lines(1))
+      end)
+
+      -- Not really valid according to the spec, but some language servers send
+      -- edits like this
```

Comment:
which ones aside from gopls ?

Reply:
None that I'm aware of, but I suspect there might be more than gopls. Should I change this comment? I don't want it to become stale if more are discovered or gopls changes.